x-pack/test/api_integration/apis/management/index_management/data_streams.ts

```diff
@@ -19,79 +19,131 @@ export default function ({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
   const es = getService('legacyEs');
 
-  const createDataStream = (name: string) => {
+  const createDataStream = async (name: string) => {
     // A data stream requires an index template before it can be created.
-    return es.dataManagement
-      .saveComposableIndexTemplate({
-        name,
-        body: {
-          // We need to match the names of backing indices with this template
-          index_patterns: [name + '*'],
-          template: {
-            mappings: {
-              properties: {
-                '@timestamp': {
-                  type: 'date',
-                },
+    await es.dataManagement.saveComposableIndexTemplate({
+      name,
+      body: {
+        // We need to match the names of backing indices with this template
+        index_patterns: [name + '*'],
+        template: {
+          mappings: {
+            properties: {
+              '@timestamp': {
+                type: 'date',
               },
             },
           },
-          data_stream: {
-            timestamp_field: '@timestamp',
-          },
         },
-      })
-      .then(() =>
-        es.dataManagement.createDataStream({
-          name,
-        })
-      );
+        data_stream: {
+          timestamp_field: '@timestamp',
+        },
+      },
+    });
+
+    await es.dataManagement.createDataStream({ name });
   };
 
-  const deleteDataStream = (name: string) => {
-    return es.dataManagement
-      .deleteDataStream({
-        name,
-      })
-      .then(() =>
-        es.dataManagement.deleteComposableIndexTemplate({
-          name,
-        })
-      );
+  const deleteComposableIndexTemplate = async (name: string) => {
```

Comment:
very minor nit; but not convinced we gain much with this function declaration.

Reply:
It was required for cleanup on this line: https://github.com/elastic/kibana/pull/70020/files#diff-1f60ccfa4a646384a55910a832d7cfd1R123. Because the deletion API only deletes the data streams, we are still left needing to clean up their related index templates after the tests run.

Reply:
I'll leave a comment in the code to make that more obvious.